crates/node/src/networking/p2p/protocol.rs

```diff
@@ -0,0 +1,43 @@
+use std::{collections::BTreeSet, net::SocketAddr};
+
+use gevulot_node::types;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) enum Handshake {
+    V1(HandshakeV1),
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) struct HandshakeV1 {
+    pub my_p2p_listen_addr: SocketAddr,
+    pub peers: BTreeSet<SocketAddr>,
+    pub http_port: Option<u16>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) enum Message {
+    V0(MessageV0),
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) enum MessageV0 {
+    Transaction(types::Transaction),
+    DiagnosticsRequest(DiagnosticsRequestKind),
+    DiagnosticsResponse(DiagnosticsResponseV0),
```

Comment:
Normally I'd put these into separate PR, but I bundled them here in order to demonstrate the thinking with these type definitions.

Normally, for now, there are only `Transaction()` kinds broadcasted in the network. No need for (request, response) pair since those aren't ever replied.

The diagnostics messages are different, however, because there the use case is such that one client can send a request where others are then expected to reply.